saas/models.py

```diff
@@ -1314,7 +1315,8 @@ def create_order(self, nb_periods, prorated_amount=0,
             ends_at = self.plan.end_of_period(self.ends_at, nb_periods)
             descr = describe_buy_periods(self.plan, ends_at, nb_periods)
             if discount_percent:
-                descr += ' - a %d%% discount' % discount_percent
+                descr += ' - a %d%% discount (code: %s)' % (
+                    discount_percent, discount_descr)
```

Comment:
If discount_descr defaults to None, then this code will write (code: None).
The idea of generalizing the discount description is good though the `create_order` parameter name is misleading. What about moving 

      ' - a %d%% discount (code: %s)' % (discount_percent, discount_descr)

to saas/views/billing.py line 420?